### DISCUSSION TITLE
bug: why must https

### DISCUSSION THREAD

**Discussion by aogg** [dec0dOS/zero-ui#147] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/8998031/220895616-c157eaca-33fb-496f-b9d4-9106a648ac38.png)

**Comment by aogg** [comment]
please allow http

**Reply by dec0dOS** [reply, marked as answer]
![Screenshot 2023-02-23 at 12 15 20](https://user-images.githubusercontent.com/14824142/220902957-3857b0c5-2965-47b0-acc3-9588831b4c15.png)
Check the docs

**Reply by gasrodriguez** [reply]
Hey @dec0dOS first of all thanks for the great work! I just started with it and looking great.

About the instructions in the doc, it may be me that I'm a docker newbie but they didn't work at the first try. Then noticed that this:
```yaml
    expose:
      - "4000"
```
should be converted to this:
```yaml
    ports:
      - "80:4000"
```
Then I can curl to it as `curl localhost/app` (strangely I didn't need to open the 80/tcp port in the firewall, in a fresh ubuntu server).

Maybe it's worth updating the doc about it?
Regards.

**Reply by dec0dOS** [reply]
Hello, @gasrodriguez. Seems to be everything is correct. You could use
```yml
 ports:
   - "4000"
```
and access your application at `localhost:4000`

**Reply by gasrodriguez** [reply]
That doesn't seem to work in my end (even after _allowing_ port 4000/tcp in the firewall):
```
$ curl localhost:4000/app
curl: (7) Failed to connect to localhost port 4000 after 0 ms: Connection refused
```
If I instead set "4000:4000" then it works (even if I _deny_ port 4000/tcp in the firewall):
```
$ curl localhost:4000/app
<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>Redirecting</title>
</head>
<body>
<pre>Redirecting to <a href="/app/">/app/</a></pre>
</body>
</html>
```

Same behavior if I open the webpage using IP:4000/app in a browser in my local network.
